### PR TITLE
Connectivity

### PR DESCRIPTION
**Pull request by charlo1998**
Connectivité intégrée au parser et fonctionnelle. Validation à compléter avec des tests

### CODE REVIEW

File: src/connectivity/Connectivity.cpp [hunk 1]
```diff
@@ -0,0 +1,370 @@
...
+
+    _face2element.resize(_nFace * 2, -1);
+    _element2face.resize(_element2elementStart[_nElem], -1);
+    _element2element.resize(_element2elementStart[_nElem], -1);
+    _face2nodeStart.resize(_nFace + 1, 0);
+    _face2node.resize(_nFace * 4, -1);
+
+
+    //creating lhelp_indice vector
+    std::vector<std::vector<int>> _lhelp_indice{{0, 2, 1, 0, 1, 3, 1, 2, 3, 2, 0, 3},
```
Comment (rouga) [inline src/connectivity/Connectivity.cpp:103]: Are these some hardcoded values ?

**Review by rouga — COMMENTED** [review]
Code seems to be working on all mesh files.
Performance-wise, its acceptable (connectivity solved in 29s for a 600k elem mesh with -03 flag)
But the compiler throws a lot of warning message. I will try to fix them before merging the code.

File: src/connectivity/Connectivity.cpp (hunk 1 above)
Comment (charlo1998) [reply src/connectivity/Connectivity.cpp:103]: this is a fixed vector depending on the vtk id of the element. its a convention on the order of the nodes and the faces

File: .gitignore [hunk 2]
```diff
@@ -5,6 +5,7 @@ cmake-build-minsizerel
 cmake-build-release
 cmake-build-relwithdebinfo
 .idea
+ONERAM6_582K.su2
```
Comment (abraham-bconstant) [inline .gitignore:8]: Pourrait être plus général avec *.su2

**Review by abraham-bconstant — CHANGES_REQUESTED** [review]
Files that should not be in the Repo:
* build.bsh
* compile_commands.json
Let me know if you need help removing them from repo without deleting them

File: src/connectivity/Connectivity.cpp [hunk 3]
```diff
@@ -0,0 +1,370 @@
...
+        }
+
+    }
+    for (size_t i = 1; i < _node2elementStart.size(); i++) {
+        _node2elementStart[i] += _node2elementStart[i - 1];
+        //printf("%2d ", _node2elementStart[i] );
+    }
+
+    int j;
+    int store[_nNode] = {0};
```
Comment (abraham-bconstant) [inline src/connectivity/Connectivity.cpp:60]: Potential Stack overflow. Change to Vector

File: src/connectivity/Connectivity.cpp [hunk 4]
```diff
@@ -0,0 +1,370 @@
...
+
+        } else {
+            std::cout << "VTK not recognised!" << '\n';
+        }
+
+
+        // ---------- commence la connectivity element2element et face ---------------
+        for (int iface = 0; iface < nLocalFacefElement; iface++) {
+            int nLocalNodefFaceI = _nNodefFace[ielem][iface];
+            int lhelp[nLocalNodefFaceI] = {0};
```
Comment (abraham-bconstant) [inline src/connectivity/Connectivity.cpp:159]: Potential stack overflow. Use vector

**Comment by abraham-bconstant** [comment]
Vous devriez utiliser clang-format! Laisser moi savoir si vous voulez que je vous montre.

File: include/mesh/Mesh.hpp [hunk 5]
```diff
@@ -90,12 +93,31 @@ namespace E3D {
...
+        void solveConnectivity();
+
 
     private:
         Parser::SU2MeshParser _parser;
         std::vector<int> _connectivity;
+        // variable calculees et assignees par connectivity
+        int nFace;
+        int nElemTot;
+        std::unique_ptr<std::vector<int>>  node2element;
```
Comment (abraham-bconstant) [inline include/mesh/Mesh.hpp:111]: Pourquoi des pointeurs de vector?

File: include/mesh/Mesh.hpp [hunk 6]
```diff
@@ -90,12 +93,31 @@ namespace E3D {
...
+
         // ------------------ Connectivity Info ----------------------
 
+        void solveConnectivity();
+
 
     private:
         Parser::SU2MeshParser _parser;
         std::vector<int> _connectivity;
+        // variable calculees et assignees par connectivity
```
Comment (abraham-bconstant) [inline include/mesh/Mesh.hpp:108]: How are these structures accessed?

File: src/connectivity/Connectivity.cpp [hunk 7]
```diff
@@ -0,0 +1,370 @@
...
+            int lpoint[_nNode] = {0};
```
Comment (abraham-bconstant) [inline src/connectivity/Connectivity.cpp:160]: Potential stack overflow

**Review by abraham-bconstant — APPROVED** [review]

**Review by rouga — APPROVED** [review]

**Pull request merged**